src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/RetryExceptionPolicyFactory.cs

```diff
@@ -14,25 +19,66 @@ namespace Microsoft.Health.Fhir.CosmosDb.Features.Storage
 {
     public class RetryExceptionPolicyFactory
     {
-        private readonly int _maxNumberOfRetries;
+        private const string RetryEndTimeContextKey = "RetryEndTime";
+        private readonly IFhirRequestContextAccessor _requestContextAccessor;
+        private readonly AsyncPolicy _sdkOnlyRetryPolicy;
+        private readonly AsyncPolicy _bundleActionRetryPolicy;
+        private readonly AsyncPolicy _backgroundJobRetryPolicy;
 
-        public RetryExceptionPolicyFactory(CosmosDataStoreConfiguration configuration)
+        public RetryExceptionPolicyFactory(CosmosDataStoreConfiguration configuration, IFhirRequestContextAccessor requestContextAccessor)
         {
+            _requestContextAccessor = requestContextAccessor;
             EnsureArg.IsNotNull(configuration, nameof(configuration));
+            EnsureArg.IsNotNull(requestContextAccessor, nameof(requestContextAccessor));
 
-            _maxNumberOfRetries = configuration.RetryOptions.MaxNumberOfRetries;
+            _sdkOnlyRetryPolicy = Policy.NoOpAsync();
+
+            _bundleActionRetryPolicy = configuration.IndividualBatchActionRetryOptions.MaxNumberOfRetries > 0
+                ? CreateExtendedRetryPolicy(configuration.IndividualBatchActionRetryOptions.MaxNumberOfRetries / configuration.RetryOptions.MaxNumberOfRetries, configuration.IndividualBatchActionRetryOptions.MaxWaitTimeInSeconds)
+                : (AsyncPolicy)Policy.NoOpAsync();
+
+            _backgroundJobRetryPolicy = CreateExtendedRetryPolicy(3, configuration.RetryOptions.MaxWaitTimeInSeconds * 3);
         }
 
-        public AsyncRetryPolicy CreateRetryPolicy()
+        public AsyncPolicy GetRetryPolicy() =>
+            _requestContextAccessor.FhirRequestContext switch
+            {
+                null => _backgroundJobRetryPolicy,
```

Comment:
Some background jobs are setting this now https://github.com/microsoft/fhir-server/blob/main/src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs#L140

Reply:
Thanks for flagging!

Reply:
addressed here: https://github.com/microsoft/fhir-server/pull/1658/commits/8d1068f1cda03085489c099533dd0cc60d687812